### ISSUE TITLE
Bug: [no-deprecated] doesn't report on shorthand property in an object expression

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10545]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=PQKhAIAEBMFMAcBOsDGBDALra4TAFBwoA2ay4KA9gHYDOG4W9AXOPYgJbUDmA3Pvip0GAM0qVwAXnABvfOHDBg4ZPEqIMtCuuQoMxAJ7zGsFifoAafAF8BQ%2BuABGZKbONK2AC0oBXYjlV1BkcfBmhKU2oAcgxjJgwrWyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
/** @deprecated */
declare const test: string;

const foo = {
  // reports correctly
  test: test,
}

const bar = {
  // should report but doesn't
  test,
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected the rule to report, as TypeScript shows this usage as deprecated.

### Actual Result

The rule misses to report this.

**Comment by JoshuaKGoldberg** [comment]
This rule is the gift that keeps on giving 🙃
